### PR TITLE
[ARCTIC-1687] Exclude kryo dependency which in ams-core from ams-optimizer module

### LINKED ISSUE #1687: [Bug]: Flink optimizer may throw LinkageError when load KryoInstantiator
### What happened?

flink-streaming-java_2.12 contains a kryo dependency, and the optimizer module depends on the submodule ams-core, which also has a kryo dependency, so there may be a dependency conflict when running Flink Optimizer.

### Affects Versions

master

### What engines are you seeing the problem on?

Optimizer

### How to reproduce

1. enable spillt disk map when starting flink optimizer
2. enable self-optimizing

### Relevant log output

```shell
Caused by: java.lang.LinkageError: loader constraint violation: when resolving method "com.esotericsoftware.kryo.Kryo$DefaultInstantiatorStrategy.setFallbackInstantiatorStrategy(Lorg/objenesis/strategy/InstantiatorStrategy;)V" the class loader (instance of org/apache/flink/util/ChildFirstClassLoader) of the current class, com/netease/arctic/utils/SerializationUtil$KryoInstantiator, and the class loader (instance of sun/misc/Launcher$AppClassLoader) for the method's defining class, com/esotericsoftware/kryo/Kryo$DefaultInstantiatorStrategy, have different Class objects for the type org/objenesis/strategy/InstantiatorStrategy used in the signature
```

### Anything else

fix some other bugs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

fix #1687 
enhance #1650 and #1624 

## Brief change log

1. resolve kryo dependency conflict
2. catch exception when unzip rewrite input tasks from db
3. increase the storage length of rewrit_input in derby

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/server/src/main/resources/derby/ams-derby-init.sql [hunk 1]
```diff
@@ -119,7 +119,7 @@ CREATE TABLE table_optimizing_process (
     plan_time           TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
     end_time            TIMESTAMP DEFAULT NULL,
     fail_reason         VARCHAR(4096),
-    rewrite_input       BLOB(64m),
+    rewrite_input       BLOB,
```
Comment (shidayang) [inline ams/server/src/main/resources/derby/ams-derby-init.sql:122]: What's the difference between BLOB(64m) and BLOB

File: ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java [hunk 2]
```diff
@@ -58,6 +64,9 @@ public static byte[] unGzip(byte[] bytes) {
       while ((len = gzipInputStream.read(buffer)) > 0) {
         byteArrayOutputStream.write(buffer, 0, len);
       }
+    } catch (ZipException e) {
+      LOG.warn("Fail to unzip, return original bytes", e);
+      return bytes;
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java:69]: I guess this was done to be compatible with data that's not compressed with Gzip. I think such compatibility would be more appropriate to be placed in the Object2ByteArrayConvert.

Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java:69]: This method is used by Object2ByteArrayConvert and TaskFilesPersistence#loadTaskInputs. Should I add try-catch to both of them?

Comment (shidayang) [reply ams/server/src/main/java/com/netease/arctic/server/utils/CompressUtil.java:69]: yeah

**Review by shidayang — APPROVED** [review]

**Review by shidayang — APPROVED** [review]
